clients/vault/src/oracle/collector/proof_builder.rs

```diff
@@ -43,34 +45,68 @@ pub enum ProofStatus {
 	NoTxSetFound(Slot),
 }
 
+
 // handles the creation of proofs.
 // this means it will access the maps and potentially the files.
 impl ScpMessageCollector {
 	fn get_slot(&self, tx_hash: &TxHash) -> Option<Slot> {
 		self.tx_hash_map().get(tx_hash).map(|slot| *slot)
 	}
 
+	
 	/// Returns either a list of ScpEnvelopes or a ProofStatus saying it failed to retrieve a list.
 	fn get_envelopes(&self, slot: Slot) -> Result<UnlimitedVarArray<ScpEnvelope>, ProofStatus> {
-		let envelopes = self._get_envelopes(slot);
+		let envelopes =
+			self._get_envelopes(slot);
 
-		let mut vec_envelopes = vec![];
+		let mut vec_envelopes  = vec![];
 
-		let fetch_more = if envelopes.is_none() {
-			true
-		} else {
+		let fetch_more = if envelopes.is_none() { true } else {
 			vec_envelopes = envelopes.unwrap().0;
-			vec_envelopes.len() < get_min_externalized_messages(self.is_public())
+			vec_envelopes.len()  < get_min_externalized_messages(self.is_public()) 
 		};
-
-		if fetch_more {
+		
+		if fetch_more{
 			let last_slot_index = *self.last_slot_index();
 			let action_sender = self.action_sender.clone();
-			if last_slot_index - MAX_SLOT_TO_REMEMBER < slot {
-				tokio::spawn(async move {
-					action_sender.send(ActorMessage::GetScpState { missed_slot: slot }).await
-				});
-			}
+			let rw_lock = self.envelopes_map_clone();
+			tokio::spawn(async move { 
+
+				if last_slot_index - MAX_SLOT_TO_REMEMBER < slot {
+					let result = action_sender
+					.send(ActorMessage::GetScpState { missed_slot : slot })
```

Comment:
This is entirely my fault, because I just remembered when I am already working with the [122 issue](https://github.com/pendulum-chain/spacewalk/issues/122).

You could actually just pass the overlay_connection:
```rust
overlay_conn: &StellarOverlayConnection,
```
and then:
```rust
overlay_conn.send(StellarMessage::GetScpState(slot.try_into().unwrap())).await;
```
because it sends directly to the Stellar-Relay lib. 

If you only have time, you can change it. Not urgent! :ok_hand:

Reply:
or even I can change it later. Really not important.